### PR TITLE
Move CI from Travis to GCB

### LINKED ISSUE #325: Move CI completely from Travis to Google Cloud Build
GCB already does most of the work for CI and it is faster and more stable than Travis. I suggest getting rid of Travis entirely and only keeping GCB around.

Note that last time I checked, GCB is still missing the step that checks for changes in generated files:

https://github.com/project-oak/oak/blob/f4d5627dd46dbedd40f1915c224348fbef39c6d9/.travis.yml#L29

@project-oak/core let me know if anyone has any objections or suggestions.

### PR DESCRIPTION
**Pull request by ipetr0v**
This change adds `git_check_init` to `cloudbuild.yaml`, thus removing the need to use Travis.

Fixes #325

### CODE REVIEW

File: cloudbuild.yaml [hunk 1]
```diff
@@ -10,8 +10,21 @@ steps:
...
     timeout: 10m
     args: ['build', '--pull', '--cache-from=gcr.io/oak-ci/oak:latest', '--tag=gcr.io/oak-ci/oak:latest', '.']
+  # Init .git repository used by check_generated
+  # Workaround for https://github.com/GoogleCloudPlatform/cloud-builders/issues/236
+  - name: 'gcr.io/cloud-builders/git'
+    id: add_git_directory
+    entrypoint: 'bash'
+    waitFor: ['pull_image']
+    timeout: 5m
+    args:
```
Comment (tiziano88) [inline cloudbuild.yaml]: Maybe it would be more consistent to check in a separate script `git_init` and rename `check_generated` to `git_check_diff` or similar. WDYT?

File: .gcloudignore [hunk 2]
```diff
@@ -0,0 +1 @@
+!.git
```
Comment (tiziano88) [inline .gcloudignore]: Why is this necessary now? Could you add a comment?

Comment (ipetr0v) [reply .gcloudignore]: Done

File: cloudbuild.yaml (hunk 1 above)
Comment (ipetr0v) [reply cloudbuild.yaml]: Done
